### ISSUE TITLE
(URGENT) When ogc.org is not available, relaton-omg hangs

### ISSUE THREAD

**Issue by ronaldtse** [relaton/relaton-omg#7]
```
#<Thread:0x0000000134d3d878 /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/workers_pool.rb:10 run> terminated with exception (report_on_exception is true):
/Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/net/http.rb:1001:in `rescue in connect': Failed to open TCP connection to www.omg.org:443 (Operation timed out - user specified timeout) (Net::OpenTimeout)
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/net/http.rb:997:in `connect'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/net/http.rb:976:in `do_start'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/net/http.rb:965:in `start'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:323:in `open_http'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:741:in `buffer_open'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:212:in `block in open_loop'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:210:in `catch'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:210:in `open_loop'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:151:in `open_uri'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-omg-1.11.0/lib/relaton_omg/scrapper.rb:14:in `scrape_page'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-omg-1.11.0/lib/relaton_omg/omg_bibliography.rb:10:in `search'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-omg-1.11.0/lib/relaton_omg/omg_bibliography.rb:18:in `get'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-omg-1.11.0/lib/relaton_omg/processor.rb:18:in `get'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/db.rb:432:in `net_retry'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/db.rb:417:in `new_bib_entry'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/db.rb:397:in `check_bibliocache'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/db.rb:74:in `fetch'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/db.rb:110:in `block in fetch_async'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/workers_pool.rb:11:in `block (2 levels) in initialize'
/Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/socket.rb:61:in `connect_internal': Operation timed out - user specified timeout (Errno::ETIMEDOUT)
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/socket.rb:137:in `connect'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/socket.rb:642:in `block in tcp'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/socket.rb:227:in `each'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/socket.rb:227:in `foreach'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/socket.rb:632:in `tcp'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/net/http.rb:998:in `connect'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/net/http.rb:976:in `do_start'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/net/http.rb:965:in `start'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:323:in `open_http'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:741:in `buffer_open'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:212:in `block in open_loop'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:210:in `catch'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:210:in `open_loop'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/open-uri.rb:151:in `open_uri'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-omg-1.11.0/lib/relaton_omg/scrapper.rb:14:in `scrape_page'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-omg-1.11.0/lib/relaton_omg/omg_bibliography.rb:10:in `search'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-omg-1.11.0/lib/relaton_omg/omg_bibliography.rb:18:in `get'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-omg-1.11.0/lib/relaton_omg/processor.rb:18:in `get'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/db.rb:432:in `net_retry'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/db.rb:417:in `new_bib_entry'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/db.rb:397:in `check_bibliocache'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/db.rb:74:in `fetch'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/db.rb:110:in `block in fetch_async'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/relaton-1.11.0/lib/relaton/workers_pool.rb:11:in `block (2 levels) in initialize'

^C/Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/ref.rb:237:in `pop': Interrupt
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/ref.rb:237:in `block in reference_queue'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/ref.rb:236:in `each'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/ref.rb:236:in `with_object'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/ref.rb:236:in `reference_queue'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/ref.rb:229:in `reference_populate'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/ref.rb:222:in `reference'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/lists.rb:37:in `ulist'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/utils.rb:13:in `convert'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/abstract_block.rb:75:in `convert'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/list.rb:30:in `convert'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/abstract_block.rb:84:in `block in content'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/abstract_block.rb:84:in `map'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/abstract_block.rb:84:in `content'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/ref_sect.rb:18:in `block in bibliography_parse'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.12.5/lib/nokogiri/xml/builder.rb:433:in `insert'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.12.5/lib/nokogiri/xml/builder.rb:416:in `method_missing'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/ref_sect.rb:16:in `bibliography_parse'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/section.rb:88:in `block in section'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.12.5/lib/nokogiri/xml/builder.rb:334:in `initialize'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.12.5/lib/nokogiri/xml/builder.rb:292:in `new'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.12.5/lib/nokogiri/xml/builder.rb:292:in `with'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/utils.rb:34:in `noko'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/section.rb:73:in `section'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/utils.rb:13:in `convert'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/abstract_block.rb:75:in `convert'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/abstract_block.rb:84:in `block in content'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/abstract_block.rb:84:in `map'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/abstract_block.rb:84:in `content'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/document.rb:1011:in `content'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/base.rb:185:in `block in middle'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.12.5/lib/nokogiri/xml/builder.rb:433:in `insert'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.12.5/lib/nokogiri/xml/builder.rb:416:in `method_missing'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/base.rb:184:in `middle'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/base.rb:156:in `block in makexml1'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.12.5/lib/nokogiri/xml/builder.rb:334:in `initialize'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.12.5/lib/nokogiri/xml/builder.rb:292:in `new'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.12.5/lib/nokogiri/xml/builder.rb:292:in `with'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/utils.rb:34:in `noko'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/base.rb:156:in `makexml1'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/base.rb:162:in `makexml'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/base.rb:114:in `document1'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/base.rb:103:in `document'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-standoc-2.0.6/lib/metanorma/standoc/utils.rb:13:in `convert'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/document.rb:955:in `convert'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciidoctor-2.0.17/lib/asciidoctor/convert.rb:120:in `convert'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-1.4.6/lib/metanorma/input/asciidoc.rb:18:in `process'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-1.4.6/lib/metanorma/processor.rb:21:in `input_to_isodoc'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-1.4.6/lib/metanorma/compile.rb:109:in `process_input_adoc'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-1.4.6/lib/metanorma/compile.rb:92:in `process_input'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-1.4.6/lib/metanorma/compile.rb:32:in `compile'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/lib/metanorma/cli/compiler.rb:45:in `compile_file'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/lib/metanorma/cli/compiler.rb:30:in `compile'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/lib/metanorma/cli/compiler.rb:35:in `compile'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/lib/metanorma/cli/site_generator.rb:89:in `compile'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/lib/metanorma/cli/site_generator.rb:29:in `block in generate'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/lib/metanorma/cli/site_generator.rb:29:in `each'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/lib/metanorma/cli/site_generator.rb:29:in `generate'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/lib/metanorma/cli/site_generator.rb:24:in `generate'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/lib/metanorma/cli/commands/site.rb:25:in `generate'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/thor-hollaback-0.2.1/lib/thor/hollaback.rb:68:in `run'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/thor-1.0.1/lib/thor/invocation.rb:116:in `invoke'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/thor-1.0.1/lib/thor.rb:243:in `block in subcommand'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/thor-hollaback-0.2.1/lib/thor/hollaback.rb:68:in `run'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/thor-1.0.1/lib/thor/base.rb:485:in `start'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/lib/metanorma/cli.rb:34:in `start'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/exe/metanorma:25:in `block in <top (required)>'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/metanorma-cli-1.5.13/exe/metanorma:43:in `<top (required)>'
	from /Users/me/.rbenv/versions/3.1.1/bin/metanorma:25:in `load'
	from /Users/me/.rbenv/versions/3.1.1/bin/metanorma:25:in `<top (required)>'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/bundler/cli/exec.rb:58:in `load'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/bundler/cli/exec.rb:58:in `kernel_load'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/bundler/cli/exec.rb:23:in `run'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/bundler/cli.rb:484:in `exec'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/bundler/cli.rb:31:in `dispatch'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/bundler/cli.rb:25:in `start'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/libexec/bundle:48:in `block in <top (required)>'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/3.1.0/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/libexec/bundle:36:in `<top (required)>'
	from /Users/me/.rbenv/versions/3.1.1/bin/bundle:25:in `load'
	from /Users/me/.rbenv/versions/3.1.1/bin/bundle:25:in `<main>'
```

**Comment by andrew2net** [comment]
fixed in v1.11.2

**Comment by ronaldtse** [comment]
Thanks@
